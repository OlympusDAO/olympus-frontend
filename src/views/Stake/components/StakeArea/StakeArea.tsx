import { t } from "@lingui/macro";
import { Divider, Grid, Zoom } from "@material-ui/core";
import { MetricCollection, Paper } from "@olympusdao/component-library";
import { useState } from "react";
import { WalletConnectedGuard } from "src/components/WalletConnectedGuard";
import { CurrentIndex, StakingAPY, TotalValueDeposited } from "src/views/TreasuryDashboard/components/Metric/Metric";

import RebaseTimer from "./components/RebaseTimer/RebaseTimer";
import { StakeBalances } from "./components/StakeBalances";
import { StakeFiveDayYield } from "./components/StakeFiveDayYield";
import { StakeInputArea } from "./components/StakeInputArea/StakeInputArea";
import { StakeNextRebaseAmount } from "./components/StakeNextRebaseAmount";
import { StakeRebaseYield } from "./components/StakeRebaseYield";

export const StakeArea: React.FC = () => {
  const [isZoomed, setIsZoomed] = useState(false);

  return (
    <Zoom in onEntered={() => setIsZoomed(true)}>
      <Paper headerText={t`Single Stake (3, 3)`} subHeader={<RebaseTimer />}>
        <Grid>
          <MetricCollection>
            <StakingAPY className="stake-apy" />

            <TotalValueDeposited className="stake-tvl" />

            <CurrentIndex className="stake-index" />
          </MetricCollection>
        </Grid>

        <WalletConnectedGuard message="Connect your wallet to stake OHM">
          <StakeInputArea isZoomed={isZoomed} />

          <StakeBalances />

          <Divider color="secondary" />

          <StakeNextRebaseAmount />

          <StakeRebaseYield />

          <StakeFiveDayYield />
        </WalletConnectedGuard>
      </Paper>
    </Zoom>
  );
};
