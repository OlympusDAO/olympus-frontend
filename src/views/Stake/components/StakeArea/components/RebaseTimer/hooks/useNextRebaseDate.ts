import { useQuery } from "react-query";
import { NetworkId } from "src/constants";
import { STAKING_ADDRESSES } from "src/constants/addresses";
import { parseBigNumber } from "src/helpers";
import { useStaticStakingContract } from "src/hooks/useContract";

export const nextRebaseDateQueryKey = () => ["useNextRebaseDate"];

export const useNextRebaseDate = () => {
  const contract = useStaticStakingContract(STAKING_ADDRESSES[NetworkId.MAINNET], NetworkId.MAINNET);

  const key = nextRebaseDateQueryKey();
  return useQuery<Date, Error>(key, async () => {
    const secondsToRebase = await contract.secondsToNextEpoch();

    const parsedSeconds = parseBigNumber(secondsToRebase, 0);

    return new Date(Date.now() + parsedSeconds * 1000);
  });
};
