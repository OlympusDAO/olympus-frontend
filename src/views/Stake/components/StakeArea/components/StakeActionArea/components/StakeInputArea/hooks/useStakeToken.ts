import { parseUnits } from "@ethersproject/units";
import { t } from "@lingui/macro";
import { ContractReceipt } from "ethers";
import { useMutation, useQueryClient } from "react-query";
import { useDispatch } from "react-redux";
import { GOHM_ADDRESSES, OHM_ADDRESSES, SOHM_ADDRESSES, STAKING_ADDRESSES } from "src/constants/addresses";
import { useWeb3Context } from "src/hooks";
import { balanceQueryKey, useBalance } from "src/hooks/useBalance";
import { useDynamicStakingContract } from "src/hooks/useContract";
import { useTestableNetworks } from "src/hooks/useTestableNetworks";
import { error as createErrorToast, info as createInfoToast } from "src/slices/MessagesSlice";

export const useStakeToken = (toToken: "sOHM" | "gOHM") => {
  const dispatch = useDispatch();
  const client = useQueryClient();
  const networks = useTestableNetworks();
  const { address, networkId } = useWeb3Context();
  const balances = useBalance(OHM_ADDRESSES);
  const contract = useDynamicStakingContract(STAKING_ADDRESSES, true);

  return useMutation<ContractReceipt, Error, string>(
    async amount => {
      if (!amount || isNaN(Number(amount))) throw new Error(t`Please enter a number`);

      const parsedAmount = parseUnits(amount, 9);

      if (!parsedAmount.gt(0)) throw new Error(t`Please enter a number greater than 0`);

      const balance = balances[networks.MAINNET].data;

      if (!balance) throw new Error(t`Please refresh your page and try again`);

      if (parsedAmount.gt(balance)) throw new Error(t`You cannot stake more than your OHM balance`);

      if (!contract) throw new Error(t`Please switch to the Ethereum network to stake your OHM`);

      if (!address) throw new Error(t`Please refresh your page and try again`);

      const shouldRebase = toToken === "sOHM";

      const transaction = await contract.stake(address, parsedAmount, shouldRebase, true);
      return transaction.wait();
    },
    {
      onError: error => {
        dispatch(createErrorToast(error.message));
      },
      onSuccess: async () => {
        const keysToRefetch = [
          balanceQueryKey(address, OHM_ADDRESSES, networkId),
          balanceQueryKey(address, toToken === "sOHM" ? SOHM_ADDRESSES : GOHM_ADDRESSES, networkId),
        ];

        const promises = keysToRefetch.map(key => client.refetchQueries(key, { active: true }));

        await Promise.all(promises);

        dispatch(createInfoToast(t`Successfully staked OHM`));
      },
    },
  );
};
