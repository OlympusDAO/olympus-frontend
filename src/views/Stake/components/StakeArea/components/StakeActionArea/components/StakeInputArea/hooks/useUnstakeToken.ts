import { parseUnits } from "@ethersproject/units";
import { t } from "@lingui/macro";
import { ContractReceipt } from "ethers";
import { useMutation, useQueryClient } from "react-query";
import { useDispatch } from "react-redux";
import { GOHM_ADDRESSES, OHM_ADDRESSES, SOHM_ADDRESSES, STAKING_ADDRESSES } from "src/constants/addresses";
import { useWeb3Context } from "src/hooks";
import { balanceQueryKey, useBalance } from "src/hooks/useBalance";
import { useDynamicStakingContract } from "src/hooks/useContract";
import { useTestableNetworks } from "src/hooks/useTestableNetworks";
import { error as createErrorToast, info as createInfoToast } from "src/slices/MessagesSlice";

export const useUnstakeToken = (fromToken: "sOHM" | "gOHM") => {
  const dispatch = useDispatch();
  const client = useQueryClient();
  const { address, networkId } = useWeb3Context();
  const networks = useTestableNetworks();
  const contract = useDynamicStakingContract(STAKING_ADDRESSES, true);

  const addresses = fromToken === "sOHM" ? SOHM_ADDRESSES : GOHM_ADDRESSES;
  const balances = useBalance(addresses);

  return useMutation<ContractReceipt, Error, string>(
    async amount => {
      if (!amount || isNaN(Number(amount))) throw new Error(t`Please enter a number`);

      const parsedAmount = parseUnits(amount, fromToken === "gOHM" ? 18 : 9);

      if (!parsedAmount.gt(0)) throw new Error(t`Please enter a number greater than 0`);

      const balance = balances[networks.MAINNET].data;

      if (!balance) throw new Error(t`Please refresh your page and try again`);

      if (parsedAmount.gt(balance))
        throw new Error(t`You cannot unstake more than your` + ` ${fromToken} ` + t`balance`);

      if (!contract) throw new Error(t`Please switch to the Ethereum network to unstake your` + ` ${fromToken}`);

      if (!address) throw new Error(t`Please refresh your page and try again`);

      const shouldRebase = fromToken === "sOHM";

      const transaction = await contract.unstake(address, parsedAmount, true, shouldRebase);
      return transaction.wait();
    },
    {
      onError: error => {
        dispatch(createErrorToast(error.message));
      },
      onSuccess: async () => {
        const keysToRefetch = [
          balanceQueryKey(address, addresses, networkId),
          balanceQueryKey(address, OHM_ADDRESSES, networkId),
        ];

        const promises = keysToRefetch.map(key => client.refetchQueries(key, { active: true }));

        await Promise.all(promises);

        dispatch(createInfoToast(t`Successfully unstaked ` + ` ${fromToken}`));
      },
    },
  );
};
