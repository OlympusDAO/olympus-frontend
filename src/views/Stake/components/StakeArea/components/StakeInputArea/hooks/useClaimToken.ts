import { useMutation, useQueryClient } from "@tanstack/react-query";
import { ContractReceipt } from "ethers";
import toast from "react-hot-toast";
import { GOHM_ADDRESSES, OHM_ADDRESSES, SOHM_ADDRESSES, STAKING_ADDRESSES } from "src/constants/addresses";
import { trackGAEvent, trackGtagEvent } from "src/helpers/analytics/trackGAEvent";
import { balanceQueryKey, useBalance } from "src/hooks/useBalance";
import { useDynamicStakingContract } from "src/hooks/useContract";
import { useTestableNetworks } from "src/hooks/useTestableNetworks";
import { warmupQueryKey } from "src/hooks/useWarmupInfo";
import { EthersError } from "src/lib/EthersTypes";
import { useAccount } from "wagmi";

/**
 * when warmup is on, useClaimToken claims the user's stake that has been warmed up
 */
export const useClaimToken = () => {
  const client = useQueryClient();
  const { address = "" } = useAccount();
  const networks = useTestableNetworks();
  const balance = useBalance(OHM_ADDRESSES)[networks.MAINNET].data;
  const contract = useDynamicStakingContract(STAKING_ADDRESSES, true);

  return useMutation<ContractReceipt, EthersError, { toToken: string }>({
    onMutate: async ({ toToken }) => {
      if (!balance) throw new Error(`Please refresh your page and try again`);

      if (!contract) throw new Error(`Please switch to the Ethereum network to claim your warmup`);

      if (!address) throw new Error(`Please refresh your page and try again`);

      const shouldRebase = toToken === "sOHM";

      const transaction = await contract.claim(address, shouldRebase);
      return transaction.wait();
    },
    onError: error => {
      toast.error("error" in error ? error.error.message : error.message);
    },
    onSuccess: async (tx, data) => {
      trackGAEvent({
        category: "Staking",
        action: "claim",
        label: `Claim to ${data.toToken}`,
        // value: new DecimalBigNumber(data.amount, 9).toApproxNumber(),
        dimension1: tx.transactionHash,
        dimension2: address,
      });

      trackGtagEvent("Claim Test 2", {
        event_category: "Claiming",
        // value: new DecimalBigNumber(data.amount, 9).toApproxNumber(),
        address: address.slice(2),
        txHash: tx.transactionHash.slice(2),
        token: data.toToken,
      });

      const keysToRefetch = [
        balanceQueryKey(address, data.toToken === "sOHM" ? SOHM_ADDRESSES : GOHM_ADDRESSES, networks.MAINNET),
        warmupQueryKey(address, networks.MAINNET),
      ];

      const promises = keysToRefetch.map(key => client.refetchQueries([key], { type: "active" }));

      await Promise.all(promises);

      toast(`Successfully claimed warmed-up balance`);
    },
  });
};
