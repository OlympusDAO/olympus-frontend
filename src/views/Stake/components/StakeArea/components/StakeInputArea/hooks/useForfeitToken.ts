import { useMutation, useQueryClient } from "@tanstack/react-query";
import { ContractReceipt } from "ethers";
import toast from "react-hot-toast";
import { OHM_ADDRESSES, STAKING_ADDRESSES } from "src/constants/addresses";
import { trackGAEvent, trackGtagEvent } from "src/helpers/analytics/trackGAEvent";
import { balanceQueryKey } from "src/hooks/useBalance";
import { useDynamicStakingContract } from "src/hooks/useContract";
import { useTestableNetworks } from "src/hooks/useTestableNetworks";
import { useWarmupClaim, warmupQueryKey } from "src/hooks/useWarmupInfo";
import { EthersError } from "src/lib/EthersTypes";
import { useAccount } from "wagmi";

/**
 * when warmup is on, useClaimToken claims the user's stake that has been warmed up
 */
export const useForfeitToken = () => {
  const client = useQueryClient();
  const { address = "" } = useAccount();
  const networks = useTestableNetworks();
  const { data: claim } = useWarmupClaim();
  const claimBalance = claim?.sohm;
  const contract = useDynamicStakingContract(STAKING_ADDRESSES, true);

  return useMutation<ContractReceipt, EthersError>({
    mutationFn: async () => {
      if (!claimBalance) throw new Error(`Please refresh your page and try again`);

      if (!contract) throw new Error(`Please switch to the Ethereum network to forfeit your warmup`);

      if (!address) throw new Error(`Please refresh your page and try again`);

      const transaction = await contract.forfeit();
      return transaction.wait();
    },
    onError: error => {
      toast.error("error" in error ? error.error.message : error.message);
    },
    onSuccess: async (tx, data) => {
      const keysToRefetch = [
        balanceQueryKey(address, OHM_ADDRESSES, networks.MAINNET),
        warmupQueryKey(address, networks.MAINNET),
      ];

      const promises = keysToRefetch.map(key => client.refetchQueries([key], { type: "active" }));

      await Promise.all(promises);

      toast(`Successfully forfeited warmed-up balance`);

      trackGAEvent({
        category: "Staking",
        action: "forfeit",
        label: `Forfeit Staking`,
        // value: new DecimalBigNumber(data.amount, 9).toApproxNumber(),
        dimension1: tx.transactionHash,
        dimension2: address,
      });

      trackGtagEvent("Forfeit Test 2", {
        event_category: "Forfeiting",
        // value: new DecimalBigNumber(data.amount, 9).toApproxNumber(),
        address: address.slice(2),
        txHash: tx.transactionHash.slice(2),
      });
    },
  });
};
