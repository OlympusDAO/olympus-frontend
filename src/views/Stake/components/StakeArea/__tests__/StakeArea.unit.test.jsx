import { BigNumber } from "ethers";
import { DecimalBigNumber } from "src/helpers/DecimalBigNumber/DecimalBigNumber";
import { useContractAllowance } from "src/hooks/useContractAllowance";
import * as Index from "src/hooks/useCurrentIndex";
import * as useWeb3Context from "src/hooks/web3Context";
import { mockWeb3Context } from "src/testHelpers";
import { act, fireEvent, render, screen } from "src/testUtils";

import { StakeArea } from "../StakeArea";

let container;
jest.mock("src/hooks/useContractAllowance");
let data;
afterEach(() => {
  jest.clearAllMocks();
  jest.restoreAllMocks();
});

describe("<StakeArea/> Disconnected", () => {
  beforeEach(async () => {
    data = jest.spyOn(useWeb3Context, "useWeb3Context");
    data.mockReturnValue({ ...mockWeb3Context, connected: false });
    await act(async () => {
      ({ container } = render(<StakeArea />));
    });
  });
  it("should ask user to connect wallet", () => {
    expect(screen.getByText("Connect Wallet")).toBeInTheDocument();
  });
});

describe("<StakeArea/> Connected no Approval", () => {
  beforeEach(async () => {
    data = jest.spyOn(useWeb3Context, "useWeb3Context");
    data.mockReturnValue(mockWeb3Context);
    useContractAllowance.mockReturnValue({ data: BigNumber.from(0) });
    await act(async () => {
      ({ container } = render(<StakeArea />));
    });
  });
  it("should render the stake input Area when connected", async () => {
    expect(screen.getByText("Unstaked Balance")).toBeInTheDocument();
  });
  it("should display unstake approval message when clicking unstake", async () => {
    fireEvent.click(screen.getByText("Unstake"));
    expect(await screen.findByText("Approve")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});

describe("<StakeArea/> Connected with Approval", () => {
  beforeEach(async () => {
    data = jest.spyOn(useWeb3Context, "useWeb3Context");
    data.mockReturnValue(mockWeb3Context);
    useContractAllowance.mockReturnValue({ data: BigNumber.from(1000) });
    Index.useCurrentIndex = jest.fn().mockReturnValue({ data: new DecimalBigNumber("10", 9) });
    await act(async () => {
      ({ container } = render(<StakeArea />));
    });
  });
  it("should switch to gOHM when toggle is selected", async () => {
    fireEvent.click(await screen.findByRole("checkbox"));
    expect(screen.getByText("Stake to gOHM")).toBeInTheDocument();
  });

  it("gOHM conversion should appear correctly when Staking to gOHM", async () => {
    fireEvent.click(await screen.findByRole("checkbox"));
    expect(screen.getByText("Stake to gOHM")).toBeInTheDocument();
    fireEvent.input(await screen.findByRole("textbox"), { target: { value: "2" } });
    expect(screen.getByText("Stake 2 OHM â†’ 0.2 gOHM")).toBeInTheDocument();
  });
});
