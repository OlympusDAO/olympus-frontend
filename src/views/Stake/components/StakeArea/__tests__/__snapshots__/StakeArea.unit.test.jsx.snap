// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StakeArea/> Connected no Approval should display unstake approval message when clicking unstake 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-67 makeStyles-root-68  MuiPaper-elevation0 MuiPaper-rounded"
    style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
    >
      <div
        class="MuiGrid-root card-header MuiGrid-item"
      >
        <div
          class="MuiBox-root MuiBox-root-69"
        >
          <div
            class="MuiBox-root MuiBox-root-70"
          >
            <h5
              class="MuiTypography-root header-text MuiTypography-h5"
            >
              Single Stake (3, 3)
            </h5>
          </div>
          <div
            class="top-right"
          />
        </div>
        <div
          class="MuiBox-root MuiBox-root-71"
        >
          <div
            class="MuiBox-root MuiBox-root-72 rebase-timer"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                style="width: 155px;"
              />
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiGrid-root"
        >
          <div
            class="MuiBox-root MuiBox-root-73"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-flex-end"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
              >
                <div
                  class="makeStyles-root-74 stake-apy"
                >
                  <div
                    class="MuiBox-root MuiBox-root-75"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary"
                    >
                      APY
                    </h5>
                    <h4
                      class="MuiTypography-root MuiTypography-h4"
                      style="width: 100%;"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                        style="width: 100%;"
                      />
                    </h4>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
              >
                <div
                  class="makeStyles-root-74 stake-tvl"
                >
                  <div
                    class="MuiBox-root MuiBox-root-76"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary"
                    >
                      Total Value Deposited
                    </h5>
                    <h4
                      class="MuiTypography-root MuiTypography-h4"
                      style="width: 100%;"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                        style="width: 100%;"
                      />
                    </h4>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
              >
                <div
                  class="makeStyles-root-74 stake-index"
                >
                  <div
                    class="MuiBox-root MuiBox-root-77"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary"
                    >
                      Current Index
                      <div
                        class="MuiBox-root MuiBox-root-78"
                        style="font-size: 16px;"
                      >
                        <div
                          class="MuiBox-root MuiBox-root-79"
                          style="display: inline-flex; justify-content: center; align-self: center;"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root info-icon MuiSvgIcon-fontSizeSmall"
                            focusable="false"
                            style="margin: 0px 5px; font-size: 1em;"
                            viewBox="0 0 20 20"
                          >
                            <path
                              d="M 10 20 C 15.475 20 20 15.473 20 10 C 20 4.518 15.473 0 9.991 0 C 4.516 0 0 4.518 0 10 C 0 15.475 4.527 20 10 20 Z M 10 18.705 C 5.189 18.714 1.287 14.812 1.297 10.001 C 1.28 5.189 5.179 1.281 9.991 1.285 C 14.807 1.28 18.714 5.182 18.714 9.999 C 18.719 14.811 14.813 18.712 10 18.702 Z M 9.941 6.242 C 10.559 6.242 11.038 5.763 11.038 5.156 C 11.043 4.547 10.549 4.053 9.94 4.058 C 9.334 4.057 8.842 4.55 8.844 5.156 C 8.843 5.761 9.337 6.249 9.941 6.242 Z M 8.216 15.444 L 12.303 15.444 C 12.623 15.444 12.871 15.204 12.871 14.895 C 12.87 14.584 12.615 14.334 12.303 14.338 L 10.868 14.338 L 10.868 8.754 C 10.868 8.346 10.658 8.065 10.269 8.065 L 8.345 8.065 C 7.919 8.045 7.631 8.493 7.826 8.872 C 7.925 9.065 8.128 9.182 8.345 9.172 L 9.652 9.172 L 9.652 14.338 L 8.216 14.338 C 7.905 14.334 7.649 14.584 7.648 14.895 C 7.648 15.204 7.897 15.444 8.216 15.444 Z"
                            />
                          </svg>
                        </div>
                      </div>
                    </h5>
                    <h4
                      class="MuiTypography-root MuiTypography-h4"
                      style="width: 100%;"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                        style="width: 100%;"
                      />
                    </h4>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-85"
        >
          <div
            class="MuiTabs-root stake-tab-buttons"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed"
              style="overflow: hidden;"
            >
              <div
                aria-label="stake tabs"
                class="MuiTabs-flexContainer MuiTabs-centered"
                role="tablist"
              >
                <button
                  aria-label="stake-button"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Stake
                  </span>
                </button>
                <button
                  aria-label="unstake-button"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Unstake
                  </span>
                </button>
              </div>
              <span
                class="PrivateTabIndicator-root-86 PrivateTabIndicator-colorPrimary-87 MuiTabs-indicator"
                style="left: 0px; width: 0px; display: none;"
              />
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-90"
          >
            <div
              class="MuiGrid-root makeStyles-inputRow-91 MuiGrid-container"
            >
              <div
                class="MuiGrid-root makeStyles-gridItem-92 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8"
              >
                <div
                  class="MuiBox-root MuiBox-root-95"
                >
                  <p
                    class="MuiTypography-root stake-note MuiTypography-body1 MuiTypography-colorTextSecondary"
                  >
                    First time unstaking
                     
                    <b>
                      sOHM
                    </b>
                    ?
                    <br />
                    Please approve Olympus DAO to use your
                     
                    <b>
                      sOHM
                    </b>
                     
                    for unstaking
                    .
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root makeStyles-gridItem-92 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
              >
                <div
                  class="MuiBox-root MuiBox-root-96"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-94 MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Approve
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiPaper-root ohm-card confirm-dialog MuiPaper-elevation0 MuiPaper-rounded"
          >
            <div
              class="MuiBox-root MuiBox-root-99"
            >
              <label
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  sOHM
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <span
                    class="MuiSwitch-root stake-to-ohm-checkbox"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-100 MuiSwitch-switchBase MuiSwitch-colorPrimary"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          aria-label="stake to gohm"
                          class="PrivateSwitchBase-input-103 MuiSwitch-input"
                          type="checkbox"
                          value=""
                        />
                        <span
                          class="MuiSwitch-thumb"
                        />
                      </span>
                    </span>
                    <span
                      class="MuiSwitch-track"
                    />
                  </span>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  gOHM
                  <div
                    class="MuiBox-root MuiBox-root-104"
                    style="display: inline-flex; justify-content: center; align-self: center;"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root info-icon MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      style="margin: 0px 5px; font-size: 1em;"
                      viewBox="0 0 20 20"
                    >
                      <path
                        d="M 10 20 C 15.475 20 20 15.473 20 10 C 20 4.518 15.473 0 9.991 0 C 4.516 0 0 4.518 0 10 C 0 15.475 4.527 20 10 20 Z M 10 18.705 C 5.189 18.714 1.287 14.812 1.297 10.001 C 1.28 5.189 5.179 1.281 9.991 1.285 C 14.807 1.28 18.714 5.182 18.714 9.999 C 18.719 14.811 14.813 18.712 10 18.702 Z M 9.941 6.242 C 10.559 6.242 11.038 5.763 11.038 5.156 C 11.043 4.547 10.549 4.053 9.94 4.058 C 9.334 4.057 8.842 4.55 8.844 5.156 C 8.843 5.761 9.337 6.249 9.941 6.242 Z M 8.216 15.444 L 12.303 15.444 C 12.623 15.444 12.871 15.204 12.871 14.895 C 12.87 14.584 12.615 14.334 12.303 14.338 L 10.868 14.338 L 10.868 8.754 C 10.868 8.346 10.658 8.065 10.269 8.065 L 8.345 8.065 C 7.919 8.045 7.631 8.493 7.826 8.872 C 7.925 9.065 8.128 9.182 8.345 9.172 L 9.652 9.172 L 9.652 14.338 L 8.216 14.338 C 7.905 14.334 7.649 14.584 7.648 14.895 C 7.648 15.204 7.897 15.444 8.216 15.444 Z"
                      />
                    </svg>
                  </div>
                </div>
              </label>
              <div
                class="MuiBox-root MuiBox-root-105"
              />
            </div>
          </div>
        </div>
        <div
          class="makeStyles-root-106 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-108"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Unstaked Balance
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            id="user-balance"
            style="text-align: right;"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80px;"
            />
          </p>
        </div>
        <div
          class="MuiPaper-root MuiAccordion-root stake-accordion Mui-expanded MuiPaper-elevation0"
        >
          <div
            aria-disabled="false"
            aria-expanded="true"
            class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content Mui-expanded"
            >
              <div
                class="makeStyles-root-106 data-row"
              >
                <div
                  class="MuiBox-root MuiBox-root-109"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                  >
                    Total Staked Balance
                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                  id="user-staked-balance"
                  style="text-align: right;"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                    style="width: 80px;"
                  />
                </p>
              </div>
            </div>
            <div
              aria-disabled="false"
              aria-hidden="true"
              class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root stake-expand MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 20 20"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-entered"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <div
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root"
                  >
                    <div
                      class="makeStyles-root-106 data-row"
                      style="padding-left: 10px;"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-110"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                        >
                          sOHM
                        </p>
                      </div>
                      <p
                        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                        id="sohm-balance"
                        style="text-align: right;"
                      >
                        <span
                          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                          style="width: 80px;"
                        />
                      </p>
                    </div>
                    <div
                      class="makeStyles-root-106 data-row"
                      style="padding-left: 10px;"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-111"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                        >
                          gOHM
                        </p>
                      </div>
                      <p
                        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                        style="text-align: right;"
                      >
                        <span
                          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                          style="width: 80px;"
                        />
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <hr
          class="MuiDivider-root"
          color="secondary"
        />
        <div
          class="makeStyles-root-106 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-112"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Next Reward Amount
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80px;"
            />
          </p>
        </div>
        <div
          class="makeStyles-root-106 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-113"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Next Reward Yield
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80px;"
            />
          </p>
        </div>
        <div
          class="makeStyles-root-106 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-114"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              ROI (5-Day Rate)
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80px;"
            />
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
